### ISSUE TITLE
Song not Playing 

### ISSUE THREAD

**Issue by suryasubhrajit** [DanielSevillano/music-you#37]
Some of the songs are not playing in the app showing error title: "**The returned video id doesn't match the requested one**". There must be a bug please solve this.

The screen recording of the issue:
https://drive.google.com/file/d/1WrSdF0IsdRL0X3SXiM4X-XqHfkF8fqZO/view?usp=drivesdk

**Comment by DanielSevillano** [comment]
Hi! The error is caused by new YouTube restrictions and it affects every FOSS YTM player, you can read more about it [here](https://github.com/25huizengek1/ViTune/issues/499). I've tried solving it with [`3baf6a3`](https://github.com/DanielSevillano/music-you/commit/3baf6a33734d6e05c76dedbbe11f806dfd1b15f7) and it works for me. There's a new version available with this fix, please let me know if it works for you!

**Comment by DanielSevillano** [comment]
I'll close this issue for now, but feel free to reopen it if something doesn't work as expected.

**Comment by suryasubhrajit** [comment]
Again the same error occurred on many songs in multiple languages, that shows "The return video ID doesn't match with the requested one".
I can't understand why this same error happening again and again...

![Image](https://github.com/user-attachments/assets/5e0badbd-6ea1-464e-9e56-ed6dbbcb3aa4)

**Comment by DanielSevillano** [comment]
I'm sorry that you're experiencing this issue again. I suppose it might also be caused by YouTube restrictions, but I can't reproduce the issue.
I've tried playing this song twice. The first time, the error appeared, but it worked after I clicked the retry button. After clearing the cache, I tried a second time and it worked fine.

**Comment by DanielSevillano** [comment]
I have implemented a fix, I'll release a new version tomorrow 🥳

**Comment by DanielSevillano** [comment]
I've released a new version, please let me know if everything works properly.

**Comment by patxixi** [comment]
Works fine, thanks for this new release!

**Comment by SuperbDhruv** [comment]
This problem has returned once again (cannot play new songs.....the old downloaded ones play just fine)

**Comment by suryasubhrajit** [comment]
I think you should fix it by bypass the age restriction or make a fake login system where the age will be 18+ for everyone. like in-built default user account with age, no need to add traditional email based login

**Comment by suryasubhrajit** [comment]
Why does the issue occurred again

**Comment by SuperbDhruv** [comment]
Nevermind it's working fine now. (😂) 

(Will update if something goes wrong again)

**Comment by DanielSevillano** [comment]
> Why does the issue occurred again

Unfortunately, these issues are caused by YouTube restrictions and they will continue affecting unofficial YTM players.

**Comment by suryasubhrajit** [comment]
so can you just put different music api along with yt to make sure all songs can be available in the app, if one api not working then it will call another?

**Comment by DanielSevillano** [comment]
Using another API that accesses YouTube Music content wouldn’t solve the issue, because it would still be affected by YouTube’s restrictions.

**Comment by DanielSevillano** [comment]
> Nevermind it's working fine now. (😂)
> 
> (Will update if something goes wrong again)

It's working for me too, I'll close the issue for now.

**Comment by DanielSevillano** [comment]
It isn't working for me again 😔

**Comment by suryasubhrajit** [comment]
can you tell me the root cause behind it??
is this for YouTube's API working over age restriction or anything else, if that you should add a dummy login system 
or else try other api insted of YouTube

**Comment by SuperbDhruv** [comment]
I suggest researching on how revanced's services work and maybe replicate the system into music you 
(Idk how to code in kotlin otherwise i would've found a solution 😭)

**Comment by hidayat-nur** [comment]
I saw in other repos, like `harmony`, it runs normally without any errors like this, I have been using it for several months.
but I don't really understand how they implement it

**Comment by SuperbDhruv** [comment]
No harmony also had stopped and required frequent updates to work (used that on PC idk bout phone)

**Comment by Diovertiiiii** [comment]
The outertune app could work, it is updated often and I think if you could update the api to this app it would be very good.

**Comment by SuperbDhruv** [comment]
i beleive outertune and Revanced youtube music share a similar api as they both are based on yt music directly.

**Comment by Diovertiiiii** [comment]
And Rimusic?

**Comment by suryasubhrajit** [comment]
Yes RiMusic also working perfectly without this bug, but I don't like that app. Please fix the bug as soon as possible.
And introduce cloud based update system which will update the app in background without need to download the apk.

**Comment by Diovertiiiii** [comment]
The same thing happens to me, Ri Music doesn't have those bugs, but it's very poorly optimized. It's a thousand times better by design. I hope they fix the bug quickly.

**Comment by suryasubhrajit** [comment]
issue didn't fixed yet.. please solve it.. songs are not playing

**Comment by DanielSevillano** [comment]
I'm working on a rewrite of the InnerTube module based on OuterTune.

**Comment by Diovertiiiii** [comment]
Please and a suggestion you should make a telegram or discord group to collect requests and problems

**Comment by suryasubhrajit** [comment]
hy daniel, is there any update to solve the bug?

**Comment by DanielSevillano** [comment]
I have implemented a fix 🥳. I'll try to release a new version today.

**Comment by Diovertiiiii** [comment]
Yes, thanks Daniel lofiu

**Comment by DanielSevillano** [comment]
I've released a new version with the fix, I hope it works for you.

**Comment by SuperbDhruv** [comment]
Some songs play but some don't

Overall update is fine

**Comment by Diovertiiiii** [comment]
There are times when the bug continues to occur but it is no longer as common as before.
